ast/src/main/scala/jawn/ast/JValue.scala

```diff
@@ -143,7 +145,7 @@ object JNum { self =>
 
   final def hybridEq(x: Long, y: Double): Boolean = {
     val z = x.toDouble
-    y == z && z.toLong == x
+    y == z && y.toLong == x
```

Comment:
Intuitively it seems like z.ToLong should always be equal to x as z was created from x. I thought a better comparison would be to compare it to y.toLong. Please correct me if I am wrong.